src/services/user.ts

```diff
@@ -13,14 +15,38 @@ class UserService {
     this.userRepository = userRepository;
   }
 
-  async getUser(id: number) {
+  async getUser(idx: number): Promise<{ id: string } & UpdateInfo> {
     try {
-      const user = await this.userRepository.findById(id);
+      const user = await this.userRepository.findByIdx(idx);
       if (!user) {
         throw new ErrorResponse(commonError.unauthorized);
       }
-      return user;
+      const { id, createdAt, updatedAt } = user;
+      return { id, createdAt, updatedAt };
     } catch (e) {
+      if ((e as ErrorResponse)?.isOperational) {
+        throw e;
+      }
+      throw new ErrorResponse(commonError.wrong);
```

Comment:
```suggestion
    } catch (e) {
      if (e instanceof ErrorResponse && e.isOperational) {
        throw e;
      }
      throw new ErrorResponse(commonError.wrong);
```

이 부분 혹시 이렇게 해도 조건문에 안 걸리나용?

Reply:
> 직접적인 에러 객체 대신 `something went very wrong!!`을 보내주고 있어서 에러를 확인할 무언가의 방법이 필요해서 넣어줬습니다. 그런데 생각해보니까 에러 미들웨어에서 development 모드일 때만 raw한 에러를 확인할 수 있게 처리를 해주고 있네요. service 쪽에서 raw한 에러를 굳이 ErrrorResponse로 wrapping 해주지 않아도 될 것 같아 봉이네요. 

요거 catch 문 안에 있는 내용 전체가 없어도 될 수도 있을 것 같아요. 일단 try 해보고 더 말씀드릴게요!!

```
if (e instanceof ErrorResponse && e.isOperational) {
    throw e;
}
throw new ErrorResponse(commonError.wrong);
```

Reply:
>`throw new ErrorResponse(commonError.wrong);` 이 코드 때문에 raw한 에러의 내용을 확인할 수 없어 console로 출력을 해주었었습니다. 하지만 try-catch를 없애봐도 controller 측에서 에러를 잡아 에러 미들웨어로 던져주고 있고 error-middleware에서 ErrorResponse가 아닌 에러 객체에 대해서 처리를 해주고 있어서 서비스측에서 try-catch로 raw한 에러 객체에 대해서 처리를 해줄 필요가 없네요 ㅎㅎ
>
> try-catch 쪽 코드 자체가 불필요한 것 같습니다. 

요거 수정해볼테니 한번 Postman으로 동작 체크 해주실 수 있을까요?

Reply:
아 그렇군요 !! 생각해보니 제가 validation 쪽 에러 처리할 때도 `try-catch`문을 처음에 사용했었다가 `try-catch`문을 없애니 에러 처리가 잘 작동했었는데, 똑같은 원리일 것 같네요 ㅎㅎ 좋습니다 !!